www_src/components/modal/modal.jsx

```diff
@@ -0,0 +1,50 @@
+var React = require('react');
+var dispatcher = require('../../lib/dispatcher');
+
+var Modal = React.createClass({
+  getInitialState: function () {
+    return {
+      isVisible: false
+    };
+  },
+  show: function () {
+    this.setState({
+      isVisible: true
+    });
+  },
+  hide: function () {
+    this.setState({
+      isVisible: false
+    });
+  },
+  componentDidMount: function () {
+    dispatcher.on('showModal', (event) => {
```

Comment:
I am totally down for namespacing the message names. Good call!

I'm not sure I get the need for a separate function in this case. It seems extra verbose for no clear advantage. I'm already invoking local methods that double as an explicit API for the modal. With your approach the only way to trigger the visibility is through message passing via the dispatcher. With my approach you could just call `show` or `hide` directly from a parent via refs if you wanted to. The extra dispatcher dependency is only needed in non parent-child compositions.